### ISSUE TITLE
Crypto operations on Ed25519

### ISSUE THREAD

**Issue by fire** [godotengine/godot-proposals#1787]
**Describe the project you are working on:**
3d game

**Describe the problem or limitation you are having in your project:**

I wanted to verify a Ed25519 public key signed signature for a PoolByteArray to determine if it is correct.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**

Like: https://github.com/godotengine/godot/pull/39755

Implement Ed25519:
```gdscript
func _load_and_test():
	var priv = load("user://key.key")
	var pub = load("user://key.pub")
	assert(pub.is_public_only())
	# Encryption
	var data = "Some data"
	var encrypted = crypto.encrypt(pub, data.to_utf8())
	# Decryption
	var decrypted = crypto.decrypt(priv, encrypted)
	# Signing
	var signature = crypto.sign(HashingContext.HASH_SHA256, data.sha256_buffer(), priv)
	# Verifying
	var verified = crypto.verify(HashingContext.HASH_SHA256, data.sha256_buffer(), signature, pub)
	# Checks
	assert(verified)
	assert(data.to_utf8() == decrypted)
```
**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**

To be determined.

Plumb the code for operations on Ed25519.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**

Not a few lines of script in complexity.

**Is there a reason why this should be core and not an add-on in the asset library?:**

Involves mbedtls which is a core library and is not exposed.

**Comment by jonbonazza** [comment]
I agree with this. Ill take a look at this in the next week or so, since I am already in the crypto code for hmac anyway (unless @fales has any concerns)

**Comment by Faless** [comment]
I don't like the idea of including another crypto library just for Ed25519.
If that's the only solution, it should just be a gdnative plugin.
If/when mbedtls supports it, we can expose it.
They seem to be actively working on it but it will probably require some time before it's included in an LTS release.

**Comment by jonbonazza** [comment]
Ah I missed the part about mbedtls not supporting it. Apologies.

I do think if mbedtls is stalled and basically abandoned, we should consider swapping to a different library, but thats for another proposal.

**Comment by Luzzotica** [comment]
Feels like I might be reviving this, but I would love to have built in Ed25519 functions like above to handle signing transactions from the game for blockchains. 

I personally work with Kadena, but I know that Ed25519 signing is required for multiple blockchains. It is also valuable for DH encryption.

**Comment by mikera** [comment]
FWIW, also interested in Ed25519 key generation / signatures in Godot. It's necessary for submitting transactions to various types of decentralised REST APIs such as used in Convex. 

Don't have any particular insights on the best implementation strategy, but I have used libsodium (https://github.com/jedisct1/libsodium) before with success. Maybe that is an option.
